### DISCUSSION TITLE
How to set username for redis

### DISCUSSION THREAD

**Discussion by Bobby99as** [mastodon/mastodon#24904] [Q&A]
I'm having problems getting Mastodon to connect to my external Redis database because there seems to be no configurational option to set the username

**Comment by nick773312** [comment]
Try adding this to .env 
REDIS_HOST=redis
REDIS_PORT=6379
REDIS_URL=redis://redis:6379/0

**Reply by Bobby99as** [reply]
Did not work

**Comment by renchap** [comment]
You need to put it into your `REDIS_URL` like this: `REDIS_URL=redis://user:password@redis:6379/0`

**Reply by Bobby99as** [reply]
I've tried that and it doesn't work when I restart Mastodon and go to the front page I keep getting http 500 errors. I guess it doesn't work with DO managed Redis databases

**Reply by ThisIsMissEm** [reply]
Ah, this is the old "SSL not supported by hiredis driver" issue, there's some documentation from a community instance over how they use DigitalOcean Redis with mastodon: https://docs.vmst.io/infrastructure/#redis

There's additional documentation on stunnel and redis from DigitalOcean here: https://www.digitalocean.com/community/tutorials/how-to-connect-to-managed-redis-over-tls-with-stunnel-and-redis-cli
